src/include/function/cast/cast_functions.h

```diff
@@ -118,8 +173,17 @@ struct CastToInt64 {
     static inline void operation(T& input, int64_t& result) {
         result = static_cast<int64_t>(input);
     }
+
+    static inline void operation(char* input, int64_t& result) {
```

Comment:
I think i understand why you need to put `char*&` before. You can still use `char*&` instead of overloading the template function.